cpu/atmega_common/periph/gpio.c

```diff
@@ -148,43 +148,26 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
     /* clear global interrupt flag */
     cli();
 
+#if defined(CPU_ATMEGA328P)
+    /* INT pins start at PD2 instead of at PD0 */
+    pin_num -= 2;
+#endif
+
     EIMSK |= (1 << pin_num);
```

Comment:
if I read the data sheet correctly there is only `0` and `1` allowed as pin_num for atmega328, maybe add an assert here?

Reply:
The previous assert already filters anything higher than 3 (line [138](https://github.com/RIOT-OS/RIOT/pull/7590/files/f6b6434cf21a6dc5e9b3904918eba7855361b5ec#diff-460a236bfe2d6795b0baca2b0cd0360fR138)) maybe it's a good idea to add a check the lower bound as well. Will do :+1: